plugins/field-bitmap/src/field-bitmap.js

```diff
@@ -0,0 +1,464 @@
+
+    this.editorPixels_ = [];
+    for (let r = 0; r < this.imgHeight_; r++) {
+      this.editorPixels_.push([]);
+      const rowDiv = this.createElementWithClassname_('div', 'pixelRow');
+      for (let c = 0; c < this.imgWidth_; c++) {
+        // Add the button to the UI and save a reference to it
+        const button = this.createElementWithClassname_('div', 'pixelButton');
+        this.editorPixels_[r].push(button);
+        rowDiv.appendChild(button);
+
+        // Handle clicking a pixel
+        this.bindEvent_(button, 'mousedown', () => {
+          this.onMouseDownInPixel_(r, c);
+          return true;
+        });
+
+        // Handle dragging into a pixel when mouse is down
+        this.bindEvent_(button, 'mouseenter', () => {
+          this.onMouseEnterPixel_(r, c);
+        });
+      }
+      pixelContainer.appendChild(rowDiv);
+    }
+
+    // Add control buttons below the pixel grid
+    this.addControlButton_(dropdownEditor, 'Randomize', this.randomizePixels_);
+    this.addControlButton_(dropdownEditor, 'Clear', this.clearPixels_);
+
+    this.render_();
+
+    return dropdownEditor;
+  }
+
+  /**
+   * Initializes the on-block display.
+   * @override
+   */
+  initView() {
+    this.blockDisplayPixels_ = [];
+    for (let r = 0; r < this.imgHeight_; r++) {
+      const row = [];
+      for (let c = 0; c < this.imgWidth_; c++) {
+        const square = Blockly.utils.dom.createSvgElement(
+            'rect', {
+              'x': c * PIXEL_SIZE,
+              'y': r * PIXEL_SIZE,
+              'width': PIXEL_SIZE,
+              'height': PIXEL_SIZE,
+              'fill': EMPTY_PIXEL_COLOR,
+              'fill_opacity': 1,
+            },
+            this.fieldGroup_);
+        row.push(square);
+      }
+      this.blockDisplayPixels_.push(row);
+    }
+  }
+
+  /**
+   * Updates the size of the block based on the size of the underlying image.
+   * @override
+   * @protected
+   */
+  updateSize_() {
+    {
+      const newWidth = PIXEL_SIZE * this.imgWidth_;
+      const newHeight = PIXEL_SIZE * this.imgHeight_;
+      if (this.borderRect_) {
+        this.borderRect_.setAttribute('width', String(newWidth));
+        this.borderRect_.setAttribute('height', String(newHeight));
+      }
+
+      this.size_.width = newWidth;
+      this.size_.height = newHeight;
+    }
+  }
+
+  /**
+   * Creates a button with specified text and action, and adds it to the
+   * parent view.
+   */
+  addControlButton_(parent, buttonText, onClick) {
+    const button = this.createElementWithClassname_('button', 'controlButton');
+    button.innerHTML = buttonText;
+    parent.appendChild(button);
+    this.bindEvent_(button, 'click', onClick);
+  }
+
+  /**
+   * Disposes of events belonging to the slider editor.
+   * @private
+   */
+  dropdownDispose_() {
+    for (const event of this.boundEvents_) {
+      Blockly.browserEvents.unbind(event);
+    }
+    this.sliderInput_ = null;
+  }
```

Comment:
Update this to properly dispose of the elements in the `editorPixels` array. They get [cleaned up automatically](https://github.com/google/blockly/blob/d2329f8cba661999cc7f03fcd6c0196770b05cdc/core/dropdowndiv.js#L232), but I believe they need to be dereferenced as well, otherwise there are performance impacts. Setting `this.editorPixels_ = []` should work.

Also update docs.